src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -355,17 +357,18 @@ impl<'tcx> RegionInferenceContext<'tcx> {
             self.universal_regions.is_local_free_region(fr),
             self.universal_regions.is_local_free_region(outlived_fr),
         );
-        debug!("report_error: fr_is_local={:?} outlived_fr_is_local={:?} category={:?}",
-               fr_is_local, outlived_fr_is_local, category);
 
+        debug!("report_error: fr_is_local={:?} outlived_fr_is_local={:?} fr_region={:?} \
+                outlived_fr_region={:?} category={:?}",
+               fr_is_local, outlived_fr_is_local, fr_region, outlived_fr_region, category);
         match (category, fr_is_local, outlived_fr_is_local) {
             (ConstraintCategory::Assignment, true, false) |
             (ConstraintCategory::CallArgument, true, false) =>
-                self.report_escaping_data_error(mir, infcx, mir_def_id, fr, outlived_fr,
-                                                category, span, errors_buffer),
+                self.report_escaping_data_error(mir, infcx, mir_def_id, fr, fr_region, outlived_fr,
```

Comment:
nit: I'm a little surprised that you decided to change the interfaces to these functions to pass along the `fr_region` and `outlived_fr_region` , since these routines are already getting the `fr` and `outlived_fr` from which those new arguments are computed.

 `to_error_region` an expensive routine to call? Or you just figure its simplest to thread the information down?

(To me, it strikes me as an extra invariant to maintain between the parameters...)

Reply:
`fr_region`/`outlived_fr_region` are derived from `fr`/`outlived_fr`. I'm not particularly sure if `to_error_region` is a particularly expensive routine, I just erred on the side of not computing it twice when I could just pass it as a parameter. I'm happy to change that though if you'd prefer keep the interface as minimal as possible.

Reply:
I'm certainly not going to block this PR based on that nitpick. Just thought I'd register my surprise at that part of the code.